python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -141,11 +141,19 @@ def y(self):
 
         @property
         def xy(self):
-            types = self._meta.input_types
-            offsets = self._meta.union_offsets
-            indices = offsets[types == self._type.value]
-            result = self._col.take(indices._column).leaves().values
-            return cudf.Series(result)
+            return cudf.Series(
+                self._get_current_features(self._type).leaves().values
+            )
+
+        def _get_current_features(self, type):
```

Comment:
What are features?

Reply:
A feature is a Point, MultiPoint, LineString, MultiLineString, Polygon, or MultiPolygon (or GeometryCollection).